src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -588,10 +587,6 @@ function VisualizeAppController(
               });
               // Manually insert a new url so the back button will open the saved visualization.
               $window.history.pushState({}, '', savedVisualizationParsedUrl.getRootRelativePath());
```

Comment:
@flash1293 also I noticed that this new history state doesn't trigger history listener:
https://github.com/elastic/kibana/blob/c1a40f33acc228ddfbba0541ae3ac46f6c6dc167/src/plugins/kibana_utils/public/state_management/url/kbn_url_tracker.ts#L137

So in this PR just `#/visualize/create` link is saved and opens by clicking on Visualize in nav bar.

Reply:
This is strange, because open a visualization after creating it in a dashboard working fine in my PR, but there is the case with the back button works wrong, so you could also check it.

Reply:
@sulemanof I checked again, you're right (probably when I tested before, something was cached). But a saved visualization currently opens just because of the [typo](https://github.com/elastic/kibana/pull/57307/files#diff-73617766d02ef0c5e3c9a21a76effbe6R91) (there is a check `link.linkToLastSubUrl === false`, but `linkToLastSubUrl` is undefined now due to the typo).

As for the history, I can navigate to a saved visualization by clicking twice on the back button. @flash1293 do you think it's a valid case?

Reply:
So I tested the current state and everything is working fine for me. When creating a vis within dashboard, it will store the id in the navlink (which is correct) and it will go back to the saved visualization (which is also correct IMHO).